From 700dd9c57a0cd2da6bad69ed92ecc5399dba24f9 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Sat, 23 Jul 2022 22:40:23 +0200
Subject: GCC: do not break compilation enabling BTI

Enabling standard arm control flow integrity passes the argument
-mmark-bti-property to GCC. This option is invalid and breaks
compilation.

Apparently the behavior in GCC is that adding BTI to
.note.gnu.property is implicit.

Bug: 819294
---
 BUILD.gn | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/BUILD.gn b/BUILD.gn
index 88b5352ce7..b8a9699e71 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1098,7 +1098,9 @@ config("toolchain") {
       if (v8_control_flow_integrity &&
           (!build_with_chromium || arm_control_flow_integrity == "standard")) {
         cflags += [ "-mbranch-protection=standard" ]
-        asmflags = [ "-mmark-bti-property" ]
+        if (is_clang) {
+          asmflags = [ "-mmark-bti-property" ]
+        }
       } else if (build_with_chromium && arm_control_flow_integrity == "pac") {
         # This should enable PAC only in C++ code (and no CFI in runtime
         # generated code). For details, see crbug.com/919548.
-- 
2.34.1

